zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -0,0 +1,749 @@
+    swap_gadget: SwapGadget<F>,
+}
+
+impl<F: FieldExt> ExecutionConfig<F> {
+    pub(crate) fn configure<TxTable, RwTable, BytecodeTable>(
+        meta: &mut ConstraintSystem<F>,
+        randomness: Column<Instance>,
+        fixed_table: [Column<Fixed>; 4],
+        tx_table: TxTable,
+        rw_table: RwTable,
+        bytecode_table: BytecodeTable,
+    ) -> Self
+    where
+        TxTable: LookupTable<F, 4>,
+        RwTable: LookupTable<F, 8>,
+        BytecodeTable: LookupTable<F, 4>,
+    {
+        let q_step = meta.complex_selector();
+        let qs_byte_lookup = meta.advice_column();
+        let advices = [(); STEP_WIDTH].map(|_| meta.advice_column());
+
+        let randomness = {
+            let mut expr = None;
+            meta.create_gate("Query randomness", |meta| {
+                expr = Some(meta.query_instance(randomness, Rotation::cur()));
+                vec![0.expr()]
+            });
+            expr.unwrap()
+        };
+
+        let step_curr = Step::new(meta, qs_byte_lookup, advices, false);
+        let step_next = Step::new(meta, qs_byte_lookup, advices, true);
+        let mut independent_lookups = Vec::new();
+        let mut presets_map = HashMap::new();
+
+        meta.create_gate("Constrain execution state", |meta| {
+            let q_step = meta.query_selector(q_step);
+
+            // Only one of execution_state should be enabled
+            let sum_to_one = step_curr
+                .state
+                .execution_state
+                .iter()
+                .fold(1.expr(), |acc, cell| acc - cell.expr());
+
+            // Cells representation for execution_state should be bool.
+            let bool_checks = step_curr
+                .state
+                .execution_state
+                .iter()
+                .map(|cell| cell.expr() * (1.expr() - cell.expr()));
+
+            std::iter::once(sum_to_one)
+                .chain(bool_checks)
+                .map(move |poly| q_step.clone() * poly)
+        });
+
+        // TODO: ExecutionState transition needs to be constraint to avoid
+        // transition from non-terminator to BeginTx.
+
+        // Use qs_byte_lookup as selector to do byte range lookup on each advice
+        // column. In this way, ExecutionGadget could enable the byte range
+        // lookup by enable qs_byte_lookup.
+        for advice in advices {
+            meta.lookup_any(|meta| {
+                let advice = meta.query_advice(advice, Rotation::cur());
+                let qs_byte_lookup =
+                    meta.query_advice(qs_byte_lookup, Rotation::cur());
+
+                vec![
+                    qs_byte_lookup.clone() * FixedTableTag::Range256.expr(),
+                    qs_byte_lookup * advice,
+                    0.expr(),
+                    0.expr(),
+                ]
+                .into_iter()
+                .zip(fixed_table.table_exprs(meta).to_vec().into_iter())
+                .map(|(input, table)| (input, table))
```

Comment:
```suggestion
```

Reply:
Fixed in [`3c57d83`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/3c57d83839d68064d81fea52ef9a55ec82a3ef5b)